assets/script.js

```diff
@@ -1,28 +1,45 @@
 /* eslint no-use-before-define:["error",{"functions":false}] */
 
-let library = [];
+class Library {
+  constructor() {
+    this.library = JSON.parse(localStorage.getItem('library') || '[]');
+  }
+
+  addBook(book) {
+    this.library.push(book);
```

Comment:
[Optional]
You should use array methods that doesn't mutate the array, e.g. [`concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), [`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [`slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) instead of [`push`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push), [`pop`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), [`splice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice).

Mutation hides change, which creates unexpected side effects, which can cause nasty bugs. When you enforce immutability you can keep your application architecture and mental model simple, which makes it easier to reason about your application

Reply:
For this change, the methods you suggest are not suitable for adding new elements, the best option for adding new elements to the array is with push.
https://www.w3schools.com/js/js_array_methods.asp

Reply:
You can do it like this:

```javascript
  addBook(title, author) {
    id = generateID(); 
    this.books = this.books.concat({ title, author, author });
    return this.books; 
 }
```